src/components/MultiSelect/MultiSelectDialog.tsx

```diff
@@ -25,30 +24,73 @@ export const MultiSelectDialog = chakra(
   forwardRef<HTMLDivElement, React.PropsWithChildren<MultiSelectDialogProps>>(
     (props, ref) => {
       const {
+        helperText,
         id,
         isBlockElement,
         isDefaultOpen,
+        isSearchable,
+        defaultItemsVisible,
         items,
-        labelText,
+        listOverflow,
+        buttonText,
+        closeOnBlur,
         onApply,
         onChange,
         onClear,
         onMixedStateChange,
         selectedItems,
+        showLabel,
+        showAll,
         width,
         ...rest
       } = props;
 
-      // Use NYPL Breakpoints hook to check screen size
-      const { isLargerThanMobile } = useNYPLBreakpoints();
-
       // Control the open or closed state of the MultiSelect.
-      const [isOpen, setIsOpen] = useState(isDefaultOpen);
+      const [isOpen] = useState(isDefaultOpen);
+      const [itemsList, setItemsList] = useState(items);
+      const [isDropDownOpen, setIsDropDownOpen] = useState(false);
+      const [viewAllLabel, setViewAllLabel] = useState("");
+      const [listHeight, setListHeight] = useState("");
+      const [listItemsCount, setListItemsCount] = useState(defaultItemsVisible);
+      const [defaultCheckboxes, setDefaultCheckboxes] = useState("none");
+      const [defaultChildCheckboxes, setDefaultChildCheckboxes] =
+        useState("none");
+      //const [prevIsOpen, setPrevIsOpen] = React.useState(isDropDownOpen);
+
+      React.useEffect(() => {
+        const calculateListHeight = () => {
+          if (listOverflow === "scroll") {
+            setViewAllLabel("View all");
+            setListHeight("260px");
+            setListItemsCount(defaultItemsVisible);
+          } else {
+            setViewAllLabel("View less");
+            setListHeight("");
+            setListItemsCount(items.length);
+          }
+        };
+
+        calculateListHeight();
+
+        if (listItemsCount === defaultItemsVisible) {
+          displayDefaultItems();
+        } else {
+          setItemsList(items);
+        }
+      }, []);
+
+      // Sets the selected items count on the menu button.
+      let getSelectedItemsCount;
```

Comment:
`getSelectedItemsCount` is not used anywhere and only declared once. You can refactor this to:
```
const getSelectedItemsCount = !!(selectedItems[id]?.items.length || 0);
```

A few reasons why:
- the value is a string but is being used on line 92 to compare to undefined. The result is a boolean.
- the string value itself is not being used so you can convert it to a boolean using the double exclamation mark `!!`.
- If the `selectedItems` is undefined, then the default is `0` which will convert to `false`.